### PR TITLE
Publish more data for imageDetails via DockerV2 task (#10019)

### PR DESCRIPTION
**Pull request by dabasnidhi**
* Publish more data for imageDetails via DockerV2 task

* Updating versions for other dependent tasks

* Review comments

### CODE REVIEW

File: Tasks/DockerV2/dockerpush.ts [hunk 1]
```diff
@@ -79,12 +82,14 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
...
     let promise = pushMultipleImages(connection, imageNames, tags, commandArguments, (image, commandOutput) => {
         output += commandOutput;
         outputImageName = image;
-        digest = extractDigestFromOutput(commandOutput);
-        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest);
-        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile).then((result) => {
+        digest = extractFromOutput(commandOutput, matchPatternForDigest);
+        imageSize = extractFromOutput(commandOutput, matchPatternForSize);
+        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest + "imageSize:" + imageSize);
+        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile, imageSize).then((result) => {
```
Comment (dhashrath-msft) [inline Tasks/DockerV2/dockerpush.ts]: We should ignore all failures silently when publishing to imagestore and trace it without affecting task status

File: Tasks/DockerV2/dockerpush.ts [hunk 2]
```diff
@@ -79,12 +82,14 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
...
     let outputImageName = "";
     let digest = "";
+    let imageSize = "";
     let promise = pushMultipleImages(connection, imageNames, tags, commandArguments, (image, commandOutput) => {
         output += commandOutput;
         outputImageName = image;
-        digest = extractDigestFromOutput(commandOutput);
-        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest);
-        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile).then((result) => {
+        digest = extractFromOutput(commandOutput, matchPatternForDigest);
```
Comment (dhashrath-msft) [inline Tasks/DockerV2/dockerpush.ts]: Any downside in combining Digest and Size regex into single and extract ?

Comment (dabasnidhi) [reply Tasks/DockerV2/dockerpush.ts]: Combined the two regex to one

File: Tasks/DockerV2/dockerpush.ts (hunk 1 above)
Comment (dabasnidhi) [reply Tasks/DockerV2/dockerpush.ts]: Done

File: Tasks/DockerComposeV0/task.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 0,
         "Minor": 150,
-        "Patch": 3
+        "Patch": 4
```
Comment (madhuv-msft) [inline Tasks/DockerComposeV0/task.json]: in master this has to be one more than here, if m150 has 3, then your update will 4 and in master it should be 5. please do the same changes to other tasks as well.

File: Tasks/DockerV2/dockerpush.ts [hunk 4]
```diff
@@ -79,13 +81,18 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
...
-        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest);
-        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile).then((result) => {
+        let extractedResult = extractDigestAndSizeFromOutput(commandOutput, matchPatternForDigestAndSize);
+        digest = extractedResult[0];
+        imageSize = extractedResult[1];
+        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest + "imageSize:" + imageSize);
+        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile, imageSize).then((result) => {
             tl.debug("ImageDetailsApiResponse: " + result);
+        }, (error) => {
+            tl.debug("publishToImageMetadataStore failed with error: " + error);
```
Comment (dhashrath-msft) [inline Tasks/DockerV2/dockerpush.ts]: prefer warning if it doesnt impact task result so that we know in logs if this failed.

**Pull request merged**